Upgrade to webpack 4.

pattiereaves opened the pull request.

Commit:
Working build, even if the processes aren't working.

Commit:
Fix webpack config so it will load.

Commit:
Working to the point of 'cannot find uglify plugin'

pattiereaves committed:
Refactor function to work for build and watch tasks.

pattiereaves commented on .babelrc:
```diff
@@ -14,7 +14,10 @@
             "node": "6.11.4",
-          }
+          },
+          "modules": "commonjs",
```
for some reason, this is the only way I could get the tests to work?! I don't understand why it wasn't necessary for the cli.

pattiereaves commented on CONTRIBUTING.md:
```diff
@@ -3,7 +3,7 @@
 At this point contribution guidelines are limited. This document contains information on setting up and developing locally; otherwise the primary concern for contributors is whether or not you are a member of the Alley Interactive organization (Huron's sponsor).
 
 ## Development
-There are two primary comonents to Huron: the CLI and the web (browser) script. Since each of these is used for different purposes, there is a different build pipeline for each.
```
All changes in this file are correcting typos

pattiereaves commented on package.json:
```diff
@@ -11,67 +11,69 @@
+    "handlebars": "^4.0.11",
+    "handlebars-loader": "^1.7.0",
+    "html-loader": "^0.5.5",
+    "html-webpack-plugin": "jantimon/html-webpack-plugin#webpack-4",
```
:shakes-fist: https://github.com/jantimon/html-webpack-plugin/issues/860#issuecomment-368353130

pattiereaves committed:
Remove unnecessary eslint disables.

pattiereaves committed:
Bump package version to 3.0

Breaking change with earlier versions of webpack.

Commit:
Fix semantic versioning.

goodguyry commented on .eslintrc:
```diff
@@ -30,7 +30,6 @@
   "plugins": [
     "jest"
   ],
-  // Do NOT change these rules
```
😂🙅🚫💅

goodguyry commented on config/webpack.browser.config.js:
```diff
@@ -2,40 +2,39 @@
+      },
+      {
+        test: /\.js$/,
+        // exclude: /node_modules/,
```
🤔 We don't need to exclude `node_modules` for either of these? If not, I'd say I'd say kill this comment.

pattiereaves replied:
we do, this was me debugging, I'm adding it back, good catch 😅

goodguyry commented on src/cli/actions.js:
```diff
@@ -172,9 +173,8 @@ export function deleteFile(filepath, store) {
-        chalk.red(`Could not delete: ${file.name}`)
-      );
+      console.warn(
+        chalk.red(`Could not delete: ${file.name}`));
```
Should this still be broken out in a separate line?

goodguyry commented on src/web/index.js:
```diff
@@ -265,7 +264,7 @@ export default class InsertNodes {
 
       if (prototype.length) {
-        id = prototype[0];
+        [id] = prototype;
```
🔥🎸

goodguyry commented on src/cli/server.js:
```diff
@@ -39,20 +37,14 @@ export default function startWebpack(config) {
-            info.errors
-          )
-        );
+        console.error(chalk.red(
```
Here too

goodguyry reviewed (approved):
The inconsistency in `console` statements' formatting (single-line vs. multi-line) bothers me, but it's obviously non-blocking. Sweet update! 🥇

ostowe committed:
Add travis.yml

ostowe committed:
Woops, rename to `.travis.yml`!

ostowe committed:
Fix html-webpack-plugin reference

ostowe committed:
Reinstall html-webpack-plugin

ostowe committed:
Move webpack-cli into dependencies (not devDependencies)

ostowe committed:
Move everything into dependencies for travis' sake

Commit:
Restore exclusion on node_modules.

Commit:
Put console func on same line.

Commit:
Make this more readable.

Commit:
Merge remote-tracking branch 'origin/webpack-4' into webpack-4

Commit:
Non-working attempt at redoing classnames as a single file.

pattiereaves committed:
Fix mysterious error where bundle was not injected properly.

pattiereaves committed:
Refactor so that it will use a single filepath for css modules.

The directory system not working with the WP implementation that requires a single JSON manifest.

Commit:
Fresh build.

Commit:
Code review recommendations.

Commit:
Remove commented line.

Commit:
Merge remote-tracking branch 'origin/fix-for-postcss-modules' into fix-for-postcss-modules

Commit:
Update documentation.

Commit:
Make this sound like a normal human wrote it.

Commit:
Merge pull request #112 from alleyinteractive/fix-for-postcss-modules

Refactor css modules implementation to point to one file, not a directory
